tidy3d/components/simulation.py

```diff
@@ -1009,6 +1010,38 @@ def _validate_monitor_size(self) -> None:
                 f"a maximum of {MAX_SIMULATION_DATA_SIZE_GB:.2f}GB are allowed."
             )
 
+        # Some monitors store much less data than what is needed internally. Make sure that the
+        # internal storage also does not exceed the limit.
+        check_monitor_types = [AbstractModeMonitor, SurfaceIntegrationMonitor, DiffractionMonitor]
+        for monitor in self.monitors:
+            if not isinstance(monitor, tuple(check_monitor_types)):
+                continue
+
+            mnt_kwargs = {
```

Comment:
I think a cleaner way to do this could be to add a property for monitors that returns the equivalent of the backend field monitor that would be used for that type of monitor (before processing) and then compute the storage size of that monitor here in the validator. What do you think?  This might also be useful on the backend potentially or for debugging purposes